lisp/poly-ein.el

```diff
@@ -10,10 +10,37 @@
   :set (lambda (symbol value)
          (set-default symbol value)
          (when value
-           (with-eval-after-load 'poly-ein
-             (poly-ein--decorate-functions))))
+           (if (featurep 'poly-ein)
+               (poly-ein--decorate-functions)
+             (with-eval-after-load 'poly-ein
+               (poly-ein--decorate-functions)))))
   :group 'ein)
 
+(defmacro poly-ein--remove-hook (label functions)
+  "Remove any hooks saying LABEL from FUNCTIONS"
+  `(mapc (lambda (x) (when (search ,label (symbol-name x))
+                       (remove-hook (quote ,functions) x t)))
+         ,functions))
+
+(defsubst poly-ein--neuter-markdown-mode ()
+  "Consolidate fragility here."
+  (when (eq major-mode 'markdown-mode)
+    (poly-ein--remove-hook "markdown" after-change-functions)
+    (poly-ein--remove-hook "markdown" jit-lock-after-change-extend-region-functions)
+    (poly-ein--remove-hook "markdown" window-configuration-change-hook)
+    (poly-ein--remove-hook "markdown" syntax-propertize-extend-region-functions)))
+
+(defun poly-ein--narrow-to-inner (modifier f &rest args)
+  (if (or pm-initialization-in-progress (not poly-ein-mode))
+      (apply f args)
+    (save-restriction
+      (widen)
+      (let ((range (pm-innermost-range
+                    (or (when (car args) (max (point-min) (funcall modifier (car args))))
+                        (point)))))
+        (narrow-to-region (car range) (cdr range))
+        (apply f args)))))
```

Comment:
There is pm-narrow-to-span for this purpose. "inner" normally refers to functionalty related to inner modes only. Spans can be body, head, tail, host.

Reply:
Thank you, I am aware of `pm-narrow-to-span`, but I needed to tweak the bounds with the `modifier`.